plugin.go

```diff
@@ -348,12 +351,52 @@ func tfFmt(config Config) *exec.Cmd {
 	if config.FmtOptions.Check != nil {
 		args = append(args, fmt.Sprintf("-check=%t", *config.FmtOptions.Check))
 	}
-	return exec.Command(
-		"terraform",
+	return createTerraformCommand(
+		config,
 		args...,
 	)
 }
 
+func getTerraformDataDir(config Config) string {
+	// Override terraform data dir
+	var terraformDataDir string
+	if config.TerraformDataDir != "" {
+		terraformDataDir = config.TerraformDataDir
+	} else if os.Getenv("TF_DATA_DIR") != "" {
+		terraformDataDir = os.Getenv("TF_DATA_DIR")
+	} else {
+		terraformDataDir = ".terraform"
+	}
+	return terraformDataDir
+}
+
+func createEnvironmentVariables(config Config) []string {
+	var environmentVariables []string = []string{}
+
+	terraformDataDir := getTerraformDataDir(config)
+	if terraformDataDir != ".terraform" {
+		environmentVariables = append(environmentVariables, fmt.Sprintf("TF_DATA_DIR=%s", terraformDataDir))
+	}
+	return environmentVariables
+}
+
+func createTerraformCommand(config Config, args ...string) *exec.Cmd {
+	command := exec.Command("terraform", args...)
+	environmentVariables := createEnvironmentVariables(config)
+	if len(environmentVariables) > 0 {
+		command.Env = append(os.Environ(), environmentVariables...)
```

Comment:
Everytime we run the command we are:

* Creating list of "variables", which really is only TF_DATA_DIR
* Setting the variables to the command

This required `Config` to be passed to all command executions.

I think it'd be better before initialization to figure out the `TF_DATA_DIR` and do a `os.SetEnv` of it to just make it available globally.  Requires much less code change and keeps code simpler IMO.